src/index.js

```diff
@@ -1,93 +1,97 @@
-const fs = require("fs"),
-    http = require("http"),
-    path = require("path"),
-    methods = require("methods"),
-    express = require("express"),
-    bodyParser = require("body-parser"),
-    session = require("express-session"),
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
-
-const isProduction = process.env.NODE_ENV === "production";
+// import http from 'http';
+// import fs from 'fs';
+import path from 'path';
+// import methods from 'methods';
+import express from 'express';
+import bodyParser from 'body-parser';
+import session from 'express-session';
+import cors from 'cors';
+// import passport from 'passport';
+import errorhandler from 'errorhandler';
+// import mongoose from 'mongoose';
+import morgan from 'morgan';
+import methodOverride from 'method-override';
+import debug from 'debug';
+import apiRoutes from './routes';
+import './models/User';
+
+const isProduction = process.env.NODE_ENV === 'production';
 
 // Create global app object
 const app = express();
+const consoleLog = debug('dev');
 
 app.use(cors());
 
 // Normal express config defaults
-app.use(require("morgan")("dev"));
+app.use(morgan('dev'));
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(bodyParser.json());
 
-app.use(require("method-override")());
-app.use(express.static(__dirname + "/public"));
+app.use(methodOverride());
+app.use(express.static(path.join(__dirname, '/public')));
 
 app.use(
-    session({
-        secret: "authorshaven",
-        cookie: { maxAge: 60000 },
-        resave: false,
-        saveUninitialized: false
-    })
+  session({
+    secret: 'authorshaven',
+    cookie: { maxAge: 60000 },
+    resave: false,
+    saveUninitialized: false
+  })
 );
 
 if (!isProduction) {
-    app.use(errorhandler());
+  app.use(errorhandler());
 }
 
-if (isProduction) {
-    mongoose.connect(process.env.MONGODB_URI);
-} else {
-    mongoose.connect("mongodb://localhost/conduit");
-    mongoose.set("debug", true);
-}
-
-require("./models/User");
+// if (isProduction) {
+//   mongoose.connect(process.env.MONGODB_URI);
+// } else {
+//   mongoose.connect('mongodb://localhost/conduit');
+//   mongoose.set('debug', true);
+// }
 
-app.use(require("./routes"));
+app.use(apiRoutes);
 
-/// catch 404 and forward to error handler
-app.use(function(req, res, next) {
-    const err = new Error("Not Found");
-    err.status = 404;
-    next(err);
+// / catch 404 and forward to error handler
+app.use((_req, _res, next) => {
```

Comment:
Good one @voke1  Since the req and res is not currently in use, eslint flags it as declared but not use. however, it is needed on that line to work Hence, I added underscore to ignore unused prefix.
I could add an underscore to some parameters that I don't want to delete, but leave as a reminder that I can use them later. which means it can be removed when we are ready to use it.

Reply:
Ok. good.